core/src/main/java/com/gorisse/thomas/sceneform/arcore/LightEstimate.kt

```diff
@@ -0,0 +1,263 @@
+        Config.LightEstimationMode.ENVIRONMENTAL_HDR -> lightEstimate.environmentalHdrEnvironment(
+            baseEnvironment,
+            previousEstimate as? HDREnvironment,
+            cameraExposureFactor,
+            config.environmentalHdrReflections,
+            config.environmentalHdrIrradiance,
+            config.environmentalHdrSpecularFilter
+        )
+        else -> null
+    }
+}
+
+fun LightEstimate.ambientIntensityEnvironment(
+    baseEnvironment: Environment?,
+    // ARCore's light estimation uses unit-less (relative) values while Filament uses a physically
+    // based camera model with lux or lumen values.
+    // In order to keep the "standard" Filament behavior we scale AR Core values.
+    cameraExposureFactor: Float
+) = Environment(
+    indirectLight = IndirectLight.Builder().apply {
+        baseEnvironment?.indirectLight?.reflectionsTexture?.let {
+            reflections(it)
+        }
+
+        // Sets light estimate to modulate the scene lighting and intensity. The rendered lights
+        // will use a combination of these values and the color and intensity of the lights.
+        // A value of a white colorCorrection and pixelIntensity of 1 mean that no changes are
+        // made to the light settings.
+        val colorCorrection = FloatArray(4).apply {
+            getColorCorrection(this, 0)
+        }.map {
+            // Rendering in linear space so first convert this value to linear space by rising
+            // to the power 2.2. Normalize the result by dividing it by 0.18, which is middle
+            // gray in linear space.
+            pow(it, 2.2f) / 0.18f
+        }
+        colorCorrection.let { (r, g, b, pixelIntensity) ->
+            // Scale and bias the estimate to avoid over darkening. Modulates ambient color with
+            // modulation factor.
+            // irradianceData must have at least one vector of three floats.
+            baseEnvironment?.sphericalHarmonics?.let { sphericalHarmonics ->
+                val colorCorrections = Color(r, g, b)
+                irradiance(
+                    3,
+                    FloatArray(sphericalHarmonics.size) { index ->
+                        when (index) {
+                            in 0..2 -> {
+                                // Use the RGB scale factors (components 0-2) to match the color
+                                // of the light in the scene
+                                // TODO: Should we apply the cameraExposureFactor?
+                                sphericalHarmonics[index] * colorCorrections[index]
+                            }
+                            else -> sphericalHarmonics[index]
+                        }
+                    })
+            }
+            // TODO: Should we also apply the intensity to the directional light?
+            baseEnvironment?.indirectLight?.intensity?.let { baseIntensity ->
+                intensity(baseIntensity * pixelIntensity * cameraExposureFactor)
```

Comment:
@romainguy @devbridie Should we also apply the intensity to the directional light?
The ambientIntensity seems very dark with the camera intensity scale factor and I can't figure out why.